ta/os_test/os_test.c

```diff
@@ -51,6 +51,90 @@ static TEE_Result check_returned_prop(
 	return TEE_SUCCESS;
 }
 
+static TEE_Result check_binprop_ones(size_t size, uint8_t *bbuf, size_t bblen)
+{
+	if (size != bblen)
+		goto err;
+
+	switch (size) {
+	case 4:
+		if (bbuf[3] != 0xff)
+			goto err;
+		/* FALLTHROUGH */
+	case 3:
+		if (bbuf[2] != 0xff)
+			goto err;
+		/* FALLTHROUGH */
+	case 2:
+		if (bbuf[1] != 0xff)
+			goto err;
+		/* FALLTHROUGH */
+	case 1:
+		if (bbuf[0] != 0xff)
+			goto err;
+		break;
+	default:
+		TEE_Panic(0);
+	}
+
+	return TEE_SUCCESS;
+
+err:
+	EMSG("Expected %zu bytes of 0xff, got %zu bytes:", size, bblen);
+	switch (size) {
+	case 1:
+		EMSG("\t%#"PRIx8, bbuf[0]);
+		break;
+	case 2:
+		EMSG("\t%#"PRIx8" %#"PRIx8, bbuf[0], bbuf[1]);
+		break;
+	case 3:
+		EMSG("\t%#"PRIx8" %#"PRIx8" %#"PRIx8,
+		     bbuf[0], bbuf[1], bbuf[2]);
+		break;
+	case 4:
+		EMSG("\t%#"PRIx8" %#"PRIx8" %#"PRIx8" %#"PRIx8,
+		     bbuf[0], bbuf[1], bbuf[2], bbuf[3]);
+		break;
+	default:
+		TEE_Panic(0);
+	}
+
+	return TEE_ERROR_GENERIC;
+}
+
+static TEE_Result get_binblock_property(TEE_PropSetHandle h,
+					char *nbuf,
+					uint8_t **bbuf,
+					size_t *bblen)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+
+	*bbuf = NULL;
+	*bblen = 0;
+	res = TEE_GetPropertyAsBinaryBlock(h, NULL, *bbuf, bblen);
+
+	if (res == TEE_SUCCESS && !*bblen)
+		return TEE_SUCCESS;
+
+	if (res != TEE_ERROR_SHORT_BUFFER) {
+		EMSG("TEE_GetPropertyAsBinaryBlock() szie query returned 0x%x\n",
+		     (unsigned int)res);
+		return TEE_ERROR_GENERIC;
```

Comment:
return res?

Reply:
If `res == TEE_SUCCESS`, we must report another return code, hence i think `TEE_ERROR_GENERIC` would be fine. 
Maybe `return res ? res : TEE_ERROR_GENERIC;` ?

Reply:
Ah you're right, no don't bother -- the original code is fine.